src/internal/joiExtended.ts

```diff
@@ -0,0 +1,106 @@
+import { BigNumber } from 'ethers';
+import Joi from 'joi';
+import web3Utils from 'web3-utils';
+
+export const joi = Joi
+  // validates ethereum address
+  .extend((joi: Joi.Root) => {
+    return {
+      base: joi.string(),
+      messages: {
+        'ethereumAddress.base': '{{#value}} is not a valid ethereum address'
+      },
+      type: 'ethereumAddress',
+      validate (value, helpers) {
+        if (!web3Utils.isAddress(value)) {
+          return { errors: helpers.error('ethereumAddress.base'), value, };
+        }
+      }
+    };
+  })
+
+  // // validates bigintish
+  // .extend((joi: Joi.Root) => {
+  //   return {
+  //     base: joi.alternatives().try(joi.string(), joi.number()),
+  //     messages: {
+  //       'bigIntish.base': '{{#value}} is not a valid bigIntish'
+  //     },
+  //     type: 'bigIntish',
+  //     validate (value: any, helpers: any) {
+  //       if (!isBigIntish(value)) {
+  //         return { errors: helpers.error('bigIntish.base'), value, };
+  //       }
+  //     }
+  //   };
+  // })
+.extend((joi: Joi.Root) => {
+  return {
+    base: joi.alternatives().try(joi.string(), joi.number()),
+    messages: {
+      'bigIntish.base': '{{#value}} is not a valid bigIntish',
+      'bigIntish.min': 'min not satisfied'
+    },
+    type: 'bigIntish',
+    validate (value: any, helpers: any) {
+      if (!isBigIntish(value)) {
+        return { errors: helpers.error('bigIntish.base'), value, };
+      }
+    },
+    rules: {
+      min: {
```

Comment:
good to have a function to determine min. I assume this is used for non-zero checks?

another thing we want to check for with bigint values is the max. Most of these values are uint256 in solidity, so max value is `2^256-1`. Sometimes these are smaller, such as uint64 (max `2^64-1`). I think bigint extended validation should default to max uint256, but let us set a different max if needed

Reply:
> good to have a function to determine min. I assume this is used for non-zero checks?

to forbid negative values